Tests/UnitTests/Core/EventData/MultiTrajectoryTests.cpp

```diff
@@ -40,77 +39,89 @@ CurvilinearParameters make_params() {
-}
-
 using ParVec_t = BoundParameters::ParVector_t;
 using CovMat_t = BoundParameters::CovMatrix_t;
 
+struct ParameterContainer {
+  SourceLink sl;
+  std::optional<Measurement<SourceLink, eLOC_0, eLOC_1, eQOP>> meas3d;
+  std::optional<Measurement<SourceLink, eLOC_0, eLOC_1>> meas2d;
+  std::optional<BoundParameters> predicted;
+  std::optional<BoundParameters> filtered;
+  std::optional<BoundParameters> smoothed;
+  CovMat_t jacobian;
+  double chi2;
+  double pathLength;
+};
+
 // std::pair<TrackState<SourceLink, BoundParameters>,
 // std::unique_ptr<FittableMeasurement<SourceLink>>>
-auto make_trackstate(size_t dim = 3) {
+template <typename track_state_t>
+auto make_trackstate(track_state_t& ts, TrackStatePropMask mask,
+                     size_t dim = 3) {
   auto plane = Surface::makeShared<PlaneSurface>(Vector3D{0., 0., 0.},
                                                  Vector3D{0., 0., 1.});
-  using TrackState = TrackState<SourceLink, BoundParameters>;
 
-  std::optional<TrackState> tso{std::nullopt};
   std::unique_ptr<FittableMeasurement<SourceLink>> fm;
+  ParameterContainer pc;
 
   if (dim == 3) {
     ActsMatrixD<3, 3> mCov;
-    // mCov << 1, 2, 3, 4, 5, 6, 7, 8, 9;
     mCov.setRandom();
 
     ActsVectorD<3> mPar;
-    // mPar << 2, 3, 4;
     mPar.setRandom();
     Measurement<SourceLink, eLOC_0, eLOC_1, eQOP> meas{
         plane, {}, mCov, mPar[0], mPar[1], mPar[2]};
 
     fm = std::make_unique<FittableMeasurement<SourceLink>>(meas);
 
     SourceLink sl{fm.get()};
-
-    TrackState ts{sl};
+    pc.sl = sl;
+    if (mask == TrackStatePropMask::Uncalibrated ||
+        mask == TrackStatePropMask::All) {
+      ts.uncalibrated() = sl;
+    }
 
     // "calibrate", keep original source link (stack address)
-    ts.measurement.calibrated =
-        decltype(meas){meas.referenceSurface().getSharedPtr(),
-                       sl,
-                       meas.covariance(),
-                       meas.parameters()[0],
-                       meas.parameters()[1],
-                       meas.parameters()[2]};
-    tso = ts;
+    pc.meas3d = {meas.referenceSurface().getSharedPtr(),
+                 sl,
+                 meas.covariance(),
+                 meas.parameters()[0],
+                 meas.parameters()[1],
+                 meas.parameters()[2]};
+    if (mask == TrackStatePropMask::Calibrated ||
+        mask == TrackStatePropMask::All) {
+      ts.setCalibrated(*pc.meas3d);
+    }
   } else if (dim == 2) {
     ActsMatrixD<2, 2> mCov;
-    // mCov << 1, 2, 3, 4;
     mCov.setRandom();
 
     ActsVectorD<2> mPar;
-    // mPar << 2, 3;
     mPar.setRandom();
     Measurement<SourceLink, eLOC_0, eLOC_1> meas{
         plane, {}, mCov, mPar[0], mPar[1]};
 
     fm = std::make_unique<FittableMeasurement<SourceLink>>(meas);
 
     SourceLink sl{fm.get()};
-
-    TrackState ts{sl};
+    pc.sl = sl;
+    if (mask == TrackStatePropMask::Uncalibrated ||
+        mask == TrackStatePropMask::All) {
```

Comment:
I think this should probably use the `ACTS_CHECK_BIT` macro. These are not the only two cases where the uncalibrated should be set.

Reply:
Didn't had this in mind - changed